src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int NAME_INDEX = 0;
+    private static final int NUMBER_INDEX = 1;
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private int supplyTotal = 0;
+    private int buyTotal = 0;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        supplyTotal = 0;
+        buyTotal = 0;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                String[] splitString = value.split(",");
+                if (splitString[NAME_INDEX].equals(SUPPLY)) {
+                    supplyTotal += Integer.parseInt(splitString[NUMBER_INDEX]);
+                } else if (splitString[NAME_INDEX].equals(BUY)) {
+                    buyTotal += Integer.parseInt(splitString[NUMBER_INDEX]);
+                }
+                value = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Couldn't read the file", e);
+        }
+
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(SUPPLY).append(",").append(supplyTotal)
+                .append(System.lineSeparator()).append(BUY).append(",").append(buyTotal)
+                .append(System.lineSeparator()).append(RESULT).append(",")
+                .append(supplyTotal - buyTotal);
 
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
+            bufferedWriter.write(stringBuilder.toString());
+        } catch (IOException e) {
+            throw new RuntimeException("Couldn't write to file", e);
+        }
```

Comment:
Approach is perfect, but public methods should be easy-to-understand, in your case you have three different logics inside getStatistic(), please extraxt to readFromFile(), createReport(), writeToFile()

Reply:
OK